### DISCUSSION TITLE
GPT-4: system_fingerprint error

### DISCUSSION THREAD

**Discussion by TheDino000** [srcnalt/OpenAI-Unity#90] [Q&A]
Hi,

I'm trying to use the gpt-4 model, but I get an error: 

> JsonSerializationException: Could not find member 'system_fingerprint' on object of type 'CreateChatCompletionResponse'. Path 'system_fingerprint', line 1, position 151.

I suppose CreateChatCompletionResponse doesn't have this parameter.
Should I modify OpenAI-Unity itself or can I pass this parameter now?
Thanks.

```
        var request = new CreateChatCompletionRequest
        {
            Model = "gpt-4-1106-preview",
            Temperature = 1,
            Stream = true,
            Messages = new List<OpenAI.ChatMessage>()
            {
                new ChatMessage() { Role = "system", Content = prompt },
                new ChatMessage() { Role = "user", Content = e.Text }
            },
            
        };
```

**Comment by JudicaMeAndrzej** [comment]
I'm also getting this error.

**Comment by srcnalt** [comment]
Since I do not have a OpenAI API key at the moment, I cannot test and reproduce the issue.

**Comment by jhinpan** [comment]
I am now getting this problem even when using gpt-3.5-turbo model. Any suggestion for solving advice?

**Comment by thubbardnd** [comment]
I have just started getting this error as well. I have tried with both gpt-3.5-turbo and gpt-4-1106-preview. @srcnalt Would there be anything I could do to help?

**Reply by jhinpan** [reply]
See Issue #99 or #98. It can be temporarily fixed now.

**Comment by tunyawat** [comment]
Under Unity project folder, go to...
Packages → OpenAIUnity → Runtime → DataTypes

Find CreateChatCompletionResponse struct
Add public string SystemFingerprint {get;set;}

![image](https://github.com/srcnalt/OpenAI-Unity/assets/6196069/5a44bce9-b0e5-453b-b03e-54464f3b46d6)

solved!
